compiler/rustc_expand/messages.ftl

```diff
@@ -27,6 +27,14 @@ expand_collapse_debuginfo_illegal =
 expand_count_repetition_misplaced =
     `count` can not be placed inside the inner-most repetition
 
+expand_crate_name_in_cfg_attr =
+    `crate_name` within an `#![cfg_attr]` attribute is forbidden
+    .help = this was previously accepted but is now an error
```

Comment:
What's the reason for wanting to keep this "help"? Wouldn't it be better to have an error code for it explaining why it's forbidden?

Related, but why is it a "help" instead of the more logic "note"?

Reply:
I can make it a note instead of a help. It's there for people who get confused by the error suddenly appearing.

Reply:
We don't have this kind of note anywhere in the compiler. I'm not aware of an breaking change where we included a note of that sort. Furthermore what does "previously" mean in 4/10/20 versions from now.

I my opinion we shouldn't have this note. The "Compatibility Notes" section in relnotes is exactly for that.

Reply:
Alright I'll remove it.